core/src/mast/serialization/decorator.rs

```diff
@@ -27,13 +27,9 @@ pub struct DecoratorInfo {
 impl DecoratorInfo {
     pub fn from_decorator(
         decorator: &Decorator,
-        data_builder: &mut DecoratorDataBuilder,
-        string_table_builder: &mut StringTableBuilder,
+        decorator_data_offset: DecoratorDataOffset,
     ) -> Self {
         let variant = EncodedDecoratorVariant::from(decorator);
-        let decorator_data_offset =
```

Comment:
Making things a bit more direct. To calculate decorator offset, this used to make a call to an argument passed into this function using two other arguments passed into this function.